src/helpers/createProof.ts

```diff
@@ -1,35 +1,64 @@
-import AllowListType from 'models/AllowListType'
-import ProofResult from 'models/ProofResult'
+import CreateProofParams from 'models/CreateProofParams'
+import PublicKey from 'models/PublicKey'
+import axios from 'axios'
 import checkIfProofUsedBefore from 'helpers/checkIfProofUsedBefore'
-import fetchAllHashes from 'helpers/fetchAllHashes'
-import getAllowMapInput from 'helpers/getAllowMapInput'
+import env from 'helpers/env'
+import getInput from 'helpers/getInput'
+import requestSignature from 'helpers/requestSignature'
+import unpackSignature from 'helpers/unpackSignature'
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 declare const snarkjs: any
 
-export default async function createProof(
-  token: string
-): Promise<{ proof: ProofResult; type: AllowListType }> {
-  for (const type of [
-    AllowListType.VC,
-    AllowListType.Founder,
-  ] as AllowListType[]) {
-    try {
-      const hashes = await fetchAllHashes(type)
-      const proof = await snarkjs.groth16.fullProve(
-        await getAllowMapInput(token, hashes),
-        'zk/AllowMapChecker.wasm',
-        'zk/AllowMapChecker_final.zkey'
-      )
-
-      if (await checkIfProofUsedBefore(type, proof.publicSignals[1])) {
-        return Promise.reject(new Error(`This token has already been used!`))
-      }
-      return { proof, type }
-    } catch (e) {
-      console.error(e)
-    }
+const baseURL = `${env.VITE_VERIFY_URL}/v0.2.1`
+
+export async function getEddsaPublicKey() {
+  const { data } = await axios.get<PublicKey>(
+    `${baseURL}/verify/eddsa-public-key`
+  )
+  return data
+}
+
+async function getHashes(id: number) {
+  const response = await fetch(`/trees/${id}.json`)
+
+  return response.json()
+}
+
+export default async function createFounderProof(params: CreateProofParams) {
+  const hashes = await getHashes(params.id)
```

Comment:
Should it be wrapped with `try/catch`, wdyt?

Reply:
We'll catch the error inside `useProof`